src/worker/search.worker.ts

```
@@ -0,0 +1,42 @@
+// src/search.worker.ts
+// استيراد الدوال من مساراتها المباشرة لتجنب الأخطاء
+import { search } from '../core/search';
+import { loadQuranData, loadMorphology, loadWordMap } from '../utils/loader';
+import { AdvancedSearchOptions, PaginationOptions } from '../types';
+let cache: any = null;
+
+// تعريف الواجهة لتجنب أخطاء TypeScript
+interface SearchMessage {
+query: string;
```

ESLint

[error] Insert `··`

(prettier/prettier)